### PR TITLE
feat: Add migration script for contacts

### PR DESCRIPTION
**Pull request by cedricmessiant**
See comment at the top of `migrateContacts.js` for what it does.

A few questions:
- I should probably rename `migrateContacts.js` but can't really find a good name. Maybe at least add a version `contactsMigration1.js`?
- I could add a test for `run`/`doMigrations` but I'm not sure that it worths it.
- it introduces a dependency to ACH. Maybe it should at least be a `devDependency`. @ptbrowne how do you handle this for external scripts?

### CODE REVIEW

**Comment by cedricmessiant** [comment]
[Migration des contacts](https://trello.com/c/m8TtEoQT/1869-migration-des-contacts)

File: package.json [hunk 1]
```diff
@@ -51,6 +51,7 @@
   },
   "dependencies": {
     "classnames": "2.2.6",
+    "cozy-ach": "^1.19.0",
```
Comment (Crash--) [inline package.json]: no ^

Comment (cedricmessiant) [reply package.json]: oh, sorry. Maybe we could add a pre-commit hook for this...

Comment (y-lohse) [reply package.json]: We can, but honestly I don't think we need to change this in PRs. Renovate does it automatically for us once merged.
Another trick is to use `yarn add cozy-ach -E` when adding the dependency.

**Comment by y-lohse** [comment]
I would say `devDependency ` is the way to go for this. Definitely not a regular dependency though.

File: scripts/migrateContacts.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+  const fixedContacts = contacts
+    .map(contact => {
+      const cozyResult = fixCozy(contact)
+      if (cozyResult.cozyFixed) result.cozy++
+      if (cozyResult.emptyCozyFixed) result.cozyEmpty++
+      const { emailFixed, contact: fixedContact } = fixEmail(cozyResult.contact)
+      if (emailFixed) result.email++
+
+      if (cozyResult.cozyFixed || emailFixed) {
```
Comment (y-lohse) [inline scripts/migrateContacts.js]: And if `cozyResult .emptyCozyFixed` ?

File: scripts/migrateContacts.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+  if (dryRun) {
+    logWithInstance(
+      `Would process ${contacts.length} contacts: \n[cozy]: ${
+        result.cozy
+      } would be fixed.\n[empty cozy]: ${
+        result.cozyEmpty
+      } would be fixed.\n[email]: ${result.email} would be fixed.\n[ghosts]: ${
+        ghosts.length
+      } would be deleted.`
```
Comment (y-lohse) [inline scripts/migrateContacts.js]: Not a problem, but the line breaks are weird here? It would be easier to read if they matched the `\n` I think.

**Comment by y-lohse** [comment]
> I should probably rename migrateContacts.js but can't really find a good name. Maybe at least add a version contactsMigration1.js?

Maybe we can prefix it with the date ?

> I could add a test for run/doMigrations but I'm not sure that it worths it.

I think there's a bug in the `doMigrations` logic (see comment above)… so it might be worth adding a test on that :)

**Comment by cedricmessiant** [comment]
Added a test, thanks @y-lohse there was a huge bug in the `doMigrations` script!

**Pull request merged**